src/main/java/org/cadixdev/mercury/mixin/MixinRemapperVisitor.java

```diff
@@ -364,4 +365,59 @@ private static MethodSignature convertSignature(final String name, final IMethod
         return new MethodSignature(name, new MethodDescriptor(parameters, convertType(binding.getReturnType())));
     }
 
+    private void remapPrivateMixinTarget(
+        final AST ast, final TypeDeclaration typeDeclaration, final ITypeBinding typeBinding
+    ) {
+        IAnnotationBinding[] annotations = typeBinding.getAnnotations();
+        List modifiers = typeDeclaration.modifiers();
+        for (Object modifier : modifiers) {
+            if (modifier instanceof NormalAnnotation) {
+                NormalAnnotation normalAnnotation = (NormalAnnotation) modifier;
+                for (Object raw : normalAnnotation.values()) {
+                    MemberValuePair pair = (MemberValuePair) raw;
+                    if ("targets".equals(pair.getName().getIdentifier())) {
+                        Expression targets = pair.getValue();
+                        if(targets instanceof StringLiteral){
+                            remapPrivateMixinTargetLiteral(ast, ((StringLiteral) targets));
+                        }
+                        else if (targets instanceof ArrayInitializer) {
+                            List expressions = ((ArrayInitializer) targets).expressions();
+                            for (Object expression : expressions) {
+                                if (expression instanceof StringLiteral) {
+                                    remapPrivateMixinTargetLiteral(ast, (StringLiteral) expression);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void remapPrivateMixinTargetLiteral(AST ast, StringLiteral literal) {
+        String className = literal.getLiteralValue().replace('.', '/');
+        if (!className.isEmpty()) {
+            ClassMapping<?, ?> classMapping = mappings.getTopLevelClassMapping(className)
+                .orElse(null);
+            if (classMapping != null) {
+                String remappedClassName = classMapping.getFullDeobfuscatedName();
+                replaceStringLiteral(
+                    ast, context, literal,
+                    remappedClassName.replace('/', '.')
```

Comment:
Mixin supports both `org/cadixdev/Example` and `org.cadixdev.Example`, MercuryMixin should be a best-effort to retain the code-style choosen by the project (where possible).

The way I can see this working is something along the lines of
```java
// Check for `.` rather than `/`, assuming the original name is in binary-format.
final boolean binaryFormat = !originalName.contains(".");
```

I will check with Sponge what their prefered style is here (as SpongeCommon uses both), and then we can use that as our default (in the future we could even make this configurable).

Reply:
@gabizou confirmed that Sponge's preference here is using `.`, so the check is probably best to be
```java
// Check for `/` rather than `.`, assuming the original name is that format.
final boolean binaryFormat = originalName.contains("/");
```

Reply:
Resolved by https://github.com/CadixDev/MercuryMixin/pull/8/commits/22dfa7410011604609c08296cb7a5de0f17bdee7.